File: C/6-retirement.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    char age[8], retire[8];
    int intage, intretire;

    printf("What is your current age? ");
    scanf("%8s", age);

    printf("At what age would you like to retire? ");
    scanf("%8s", retire);
    
    char curMonth[8], curDay[8], curYear[8];
    sscanf(__DATE__, "%s %s %s", curMonth, curDay, curYear);

    intage = atoi(age);
    intretire = atoi(retire);

    printf("You have %d years left until you can retire.\n", (intretire - intage));
    printf("It's %s, so you can retire in %d.\n", curYear, atoi(curYear) + (intretire - intage));
    
    return 0;
}

File: go/6-retirement.go
package main

import "fmt"

const YEAR int = 2015

func main() {
	var age, retire int

	fmt.Printf("What is your current age? ")
	fmt.Scan(&age)

	fmt.Printf("At what age would you like to retire? ")
	fmt.Scan(&retire)

	remaining := retire - age
	
	fmt.Printf("You have %d years left until you can retire.\n", remaining)
	fmt.Printf("It's %d, so you can retire in %d.\n", YEAR, YEAR + remaining)
}